transport_nantes/topicblog/models.py

```diff
@@ -1678,3 +1680,30 @@ def get_edit_url(self):
                 "topicblog:edit_panel",
                 kwargs={"pkid": self.pk, "the_slug": self.slug},
             )
+
+
+def get_random_panel():
+    """Choose a displayable panel at random."""
+    # Using objects.filter().order_by.distinct.count() would make a
+    # lot of sense, but the django ORM doesn't support distinct() on
+    # sqlite3 (in dev).  So we have to do this a bit differently
+    # unless/until we support an easy postgresql in vagrant solution
+    # for dev.
+    # num_published_panels = (
+    #     TopicBlogPanel.objects.filter(publication_date__isnull=False)
+    #     .annotate(num_published=Count("slug"))
+    #     .count()
+    # )
+    published_panels = (
+        TopicBlogPanel.objects.filter(publication_date__isnull=False)
+        .values("slug")
+        .annotate(pub_date=Max("publication_date"))
+    )
+    the_panel_values = published_panels[
+        randint(0, published_panels.count() - 1)
```

Comment:
I think Count() is actually making another query https://docs.djangoproject.com/en/4.1/ref/models/querysets/#django.db.models.query.QuerySet.count , and because you will evaluate the query's object anyway, using len() might be even more efficient, but performance isn't even an issue at this point. :)

Reply:
Yeah, this is contorted.  In this case, the dereference (`[randint]`) will cause us only to fetch one record, so I don't think we actually fully retrieve the queryset.